doc/python/developer.rst

```diff
@@ -62,4 +62,66 @@ And that's it! The check run in the CI is equivalent to here, so you should
 always be able to reproduce failures. And if you install the pre-commit hook,
 you can largely prevent them by always catching them before commit.
 
+Writing Python Commands
+-----------------------
+
+The Flux command design is done in a way that some commands are generated
+from the C code, and others are helpers that come from Python. 
+If you need to add a new command (and want it to show up) there are some tricks you need to know.
+This short guide will help you on this journey! 🏔️
+
+Locations
+~~~~~~~~~
+
+Flux commands can be found under ``src/cmd``. You'll notice many files with
+a prefix "flux-". By default, any file with this prefix will be discovered
+and available to the user as a flux command. As an example, ``flux-run.py`` will
+be available as ``flux run`` and ``flux-job.c`` will be available as ``flux job``. 
+This shows that we have a mix of C derived and Python derived commands. if you are
+interested in how this works, look at ``src/cmd/cmdhelp.c``. If you provide a file with 
+a different name here (e.g., ``supporting-file.py`` it will not be rendered to a command,
+so feel free to do that if the design affords that.
+
+Let's now say that we add a new command ``flux-foo.py``. We would expect to compile
+Flux and see it available in ``flux help``, right? Wrong!
+While the command will technically work as ``flux foo``, it won't show up in the help,
+and there are a few tricks to getting that working, discussed next.
+
+Adding a New Command
+~~~~~~~~~~~~~~~~~~~~
+
+The command generation works as follows:
+
+ - A new command should be a Python file added to ``src/cmd`` named like ``flux-<command-name>.py``.
+ - The command can use (and write new) shared base classes that are found under ``src/bindings/python/flux/cli`` 
+ - The script ``etc/gen-cmdhelp.py`` is run during build, and it looks for entries in ``doc/manpages.py``
+ - This means you need to add line entries for your new commands including the .rst file path relative to doc, e.g.:
+ 
+  .. code-block:: python
+
+    ('man1/flux-submit', 'flux-submit', 'submit a job to a Flux instance', [author], 1),
+    ('man1/flux-run', 'flux-run', 'run a Flux job interactively', [author], 1),
+    ('man1/flux-bulksubmit', 'flux-bulksubmit', 'submit jobs in bulk to a Flux instance', [author], 1),
+    ('man1/flux-alloc', 'flux-alloc', 'allocate a new Flux instance for interactive use', [author], 1),
+    ('man1/flux-batch', 'flux-batch', 'submit a batch script to Flux', [author], 1),
+
+ - And of course, the dependency to that means actually writing the file! Python commands are in "man1"
+ - Try to use shared logic whenever possible! E.g., ``doc/man1/common`` has common snippets.
+ - The script generates ``etc/flux/help.d/core.json`` where you can sanity check the output.
+ 
```

Comment:
Don't forget updating doc/Makefile.am (`MAN_FILES_PRIMARY`) and  src/cmd/Makefile.am (`dist_fluxcmd_SCRIPTS` ).